torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -144,23 +205,46 @@ ThreadPredicateMap::const_iterator ThreadPredicateMap::end() const {
   return thread_predicates_.end();
 }
 
-const ir_utils::ParallelTypeBitmap& ThreadPredicateMap::at(
+const ThreadPredicateMap::MapType::mapped_type& ThreadPredicateMap::at(
     const TensorView* tv) const {
   return thread_predicates_.at(tv);
 }
 
-ir_utils::ParallelTypeBitmap& ThreadPredicateMap::at(const TensorView* tv) {
+ThreadPredicateMap::MapType::mapped_type& ThreadPredicateMap::at(
+    const TensorView* tv) {
   return thread_predicates_.at(tv);
 }
 
-ir_utils::ParallelTypeBitmap& ThreadPredicateMap::operator[](
+ThreadPredicateMap::MapType::mapped_type& ThreadPredicateMap::operator[](
     const TensorView* tv) {
   return thread_predicates_[tv];
 }
 
+void ThreadPredicateMap::emplace(
```

Comment:
why `emplace()` vs. a simpler `push_back`? seems a premature optimization (and a bit confusion since there's one emplace overload which wraps another, defeating the point of a true emplace)..?

Reply:
Nothing particular. These are private functions, so I didn't think too much about design alternatives. I didn't realize it's not going to be true emplace. I'll make them simpler with `insert`.